### PR TITLE
Fix/Search bugs

### PR DESCRIPTION
**Pull request by richpjames**
A couple of bug fixes I spotted whilst doing the demo: 
1. The incorrect start date was being used causing a crash
3. Ensure the start date input is populated correctly

### CODE REVIEW

File: server/controllers/match/bedSearchController.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import applyPaths from '../../paths/apply'
 import { PersonService } from '../../services'
 import BedService from '../../services/bedService'
 
-import { startDateFromParams } from '../../utils/matchUtils'
```
Comment (pezholio) [inline server/controllers/match/bedSearchController.ts:10]: This doesn't seem relevant to this commit message - bad rebase?

Comment (richpjames) [reply server/controllers/match/bedSearchController.ts:10]: Yeah that commit was no longer needed

File: server/controllers/match/bedSearchController.ts [hunk 2]
```diff
@@ -37,8 +38,8 @@ export default class BedSearchController {
 
   search(): RequestHandler {
     return async (req: Request, res: Response) => {
-      const query = (Object.keys(req.query).length ? req.query : undefined) as unknown as BedSearchParametersUi
-      const body = (Object.keys(req.body).length ? req.body : undefined) as unknown as BedSearchParametersUi
+      const query = objectIfNotEmpty(req.query) as unknown as BedSearchParametersUi
```
Comment (pezholio) [inline server/controllers/match/bedSearchController.ts:41]: 👌 - nice!

Comment (pezholio) [reply server/controllers/match/bedSearchController.ts:41]: Could we add a generic to do the casting too maybe? So:

```typescript
const query = objectIfNotEmpty<BedSearchParametersUi>(req.query)
```

**Review by pezholio — APPROVED** [review]
LGTM - just one tiny suggestion, but might not actually be possible!
